src/Transit/Internal/Network.hs

```diff
@@ -111,32 +114,25 @@ getSocketPort (Just sock') = do
   portnum <- socketPort sock'
   return (Just portnum)
 
-type Hostname = Text
-
-ipv4ToHostname :: Word32 -> Hostname
-ipv4ToHostname ip =
-  let (q1, r1) = ip `divMod` 256
-      (q2, r2) = q1 `divMod` 256
-      (q3, r3) = q2 `divMod` 256
-  in
-    show r1 <> "." <> show r2 <> "." <> show r3 <> "." <> show q3
-
 buildDirectHints :: PortNumber -> IO (Set.Set ConnectionHint)
 buildDirectHints portnum = do
   nwInterfaces <- getNetworkInterfaces
   let nonLoopbackInterfaces =
-        filter (\nwInterface ->
-                   let (IPv4 addr4) = ipv4 nwInterface
-                   in
-                     (ipv4ToHostname addr4 /= "0.0.0.0")
-                     && (ipv4ToHostname addr4 /= "127.0.0.1"))
-        nwInterfaces
-  return $ Set.fromList $ map (\nwInterface ->
-                                 let (IPv4 addr4) = ipv4 nwInterface in
-                                   Direct Hint { hostname = ipv4ToHostname addr4
-                                               , port = fromIntegral portnum
-                                               , priority = 0
-                                               , ctype = DirectTcpV1 }) nonLoopbackInterfaces
+          filter (\nwInterface -> name nwInterface /= "lo" &&
+                                  (show (ipv4 nwInterface) /= ("0.0.0.0" :: Text) ||
+                                   show (ipv6 nwInterface) /= ("0:0:0:0:0:0:0:0" :: Text)))
+          nwInterfaces
+      dhints = concatMap (\nwInterface ->
+                         [ Direct Hint { hostname = show (ipv4 nwInterface)
+                                       , port = fromIntegral portnum
+                                       , priority = 0
+                                       , ctype = DirectTcpV1 }
+                         , Direct Hint { hostname = show (ipv6 nwInterface)
+                                       , port = fromIntegral portnum
+                                       , priority = 0
+                                       , ctype = DirectTcpV1 }
```

Comment:
Above we filtered the interfaces to find any that had a real address for _either_ IPv4 or IPv6 - but here we take that list and we put both IPv4 and IPv6 addresses from all interfaces into the hints.  It seems like this will put INADDR_ANY / IN6ADDR_ANY into the hints for some interfaces?

Reply:
`DirectTcpV1` can have either kind of address -- if that's the question :)

Reply:
That part sounds good.  I think this code will put "0.0.0.0" into a `DirectTcpV1` hint sometimes though - which seems wrong.

Reply:
oh, yeah that sounds wrong to me too